cpp/src/text/ngrams_tokenize.cu

```diff
@@ -188,21 +182,17 @@ std::unique_ptr<cudf::column> ngrams_tokenize(cudf::strings_column_view const& s
 
   // compute the number of ngrams per string to get the total number of ngrams to generate
   // Ex. ngram-counts = [2,1]; ngram-offsets = [0,2,3]; total = 3 bigrams
-  rmm::device_uvector<cudf::size_type> ngram_offsets(strings_count + 1, stream);
-  auto d_ngram_offsets = ngram_offsets.data();
-  thrust::transform_inclusive_scan(
-    rmm::exec_policy(stream),
-    thrust::make_counting_iterator<cudf::size_type>(0),
-    thrust::make_counting_iterator<cudf::size_type>(strings_count),
-    d_ngram_offsets + 1,
+  auto const ngram_counts = cudf::detail::make_counting_transform_iterator(
+    0,
     cuda::proclaim_return_type<cudf::size_type>(
       [d_token_offsets, ngrams] __device__(cudf::size_type idx) {
-        auto token_count = d_token_offsets[idx + 1] - d_token_offsets[idx];
+        auto token_count =
+          static_cast<cudf::size_type>(d_token_offsets[idx + 1] - d_token_offsets[idx]);
```

Comment:
I think I was expecting the number of ngrams/tokens to be `int64_t` instead of `size_type`. It feels more like the character count (arbitrarily large). Can you help me understand this? Is there a second set of changes needed, after migrating to the offsetalator?

Reply:
The number of tokens/ngrams becomes the number of _rows_ in the output column and the number of rows still cannot exceed `size_type` even if the total size of the strings buffer is very large. So the `token_offsets` is **not** a byte-offset and so will always contain values less than `max(size_type)`.

Reply:
Got it! I didn’t know it turned into a row count.